### PR TITLE
arrange for the saving of a new task to the db

### PR DESCRIPTION
**Pull request by dvnrsn**

### CODE REVIEW

File: src/modal/modal-styles.css [hunk 1]
```diff
@@ -13,10 +13,9 @@
 
 .modal-content {
   background-color: #242f34;
-  box-shadow: inset 0px 0px 40px 40px #dba632;
```
Comment (dvnrsn) [inline src/modal/modal-styles.css:16]: meh, don't like the look

File: src/modal/modal-styles.css [hunk 2]
```diff
@@ -13,10 +13,9 @@
...
   margin: auto;
   padding: 20px;
-  width: 60%;
+  width: 500px;
```
Comment (dvnrsn) [inline src/modal/modal-styles.css:18]: modal shouldn't be responsive. just fixed width

File: src/modal/modal-styles.css [hunk 3]
```diff
@@ -66,3 +65,14 @@ button {
...
+.days, .days > * {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 30px;
+}
+
+.days > * {
+  margin: 0px 4px;
+}
```
Comment (dvnrsn) [inline src/modal/modal-styles.css:78]: trying to get some spacing on the days in the modal

File: src/modal/modal.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
 import React from "react";
 import "./modal-styles.css";
 import { days as daysOfWeek } from "../helper.js";
+import useSaveTask from './use-save-task';
```
Comment (dvnrsn) [inline src/modal/modal.js:4]: this custom hook is responsible for firing off a request to save task

File: src/modal/modal.js [hunk 5]
```diff
@@ -31,26 +37,40 @@ export function Modal(props) {
...
+                        id={d}
+                        name={d}
+                        type="checkbox"
+                        checked={days.includes(d)}
+                        onChange={handleCheckbox}
+                      />
+                      <label htmlFor={d}>{d}</label>
+                    </div>
+                  ))}
+                </div>
```
Comment (dvnrsn) [inline src/modal/modal.js:59]: split up like this for SMT /new line WThFS

File: src/modal/use-save-task.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        body: JSON.stringify(data),
+      })
+        .then((res) => res.json())
+        .then((task) => {
+          task.days = JSON.parse(task.days)
+          cb(task)
+        });
+    }
+  }, [cb, task, submit]);
+}
```
Comment (dvnrsn) [inline src/modal/use-save-task.js:26]: the magic happens here

Comment (tcovington1) [reply src/modal/use-save-task.js:26]: This is awesome, great job! I'm loving these custom hooks! Something I need to get better at.

**Pull request merged**